librosa/display.py

```diff
@@ -1220,8 +1323,197 @@ def __coord_time(n, sr=22050, hop_length=512, **_kwargs):
     return axes_same_and_not_none and axes_same_lim
+
+
+def waveshow(
+    y,
+    sr=22050,
+    max_points=11025,
+    x_axis="time",
+    offset=0.0,
+    marker="",
+    where="post",
+    label=None,
+    ax=None,
+    **kwargs,
+):
+    """Visualize a waveform in the time domain.
+
+    This function constructs a plot which adaptively switches between a raw
+    samples-based view of the signal (`matplotlib.pyplot.step`) and an
+    amplitude-envelope view of the signal (`matplotlib.pyplot.fill_between`)
+    depending on the time extent of the plot's viewport.
+
+    More specifically, when the plot spans a time interval of less than `max_points /
+    sr` (by default, 1/2 second), the samples-based view is used, and otherwise a
+    downsampled amplitude envelope is used. 
+    This is done to limit the complexity of the visual elements to guarantee an
+    efficient, visually interpretable plot.
+
+    When using interactive rendering (e.g., in a Jupyter notebook or IPython
+    console), the plot will automatically update as the view-port is changed, either
+    through widget controls or programmatic updates.
+
+
+    Parameters
+    ----------
+    y : np.ndarray [shape=(n,) or (2,n)]
+        audio time series (mono or stereo)
+
+    sr : number > 0 [scalar]
+        sampling rate of ``y``
+
+    max_points : postive integer
+        Maximum number of samples to draw.  When the plot covers a time extent
+        smaller than `max_points / sr` (default: 1/2 second), samples are drawn.
+
+        If drawing raw samples would exceed `max_points`, then a downsampled
+        amplitude envelope extracted from non-overlapping windows of `y` is
+        visualized instead.  The parameters of the amplitude envelope are defined so
+        that the resulting plot cannot produce more than `max_points` frames.
+
+    x_axis : str or None
+        Display of the x-axis ticks and tick markers. Accepted values are:
+
+        - 'time' : markers are shown as milliseconds, seconds, minutes, or hours.
+                    Values are plotted in units of seconds.
+
+        - 's' : markers are shown as seconds.
+
+        - 'ms' : markers are shown as milliseconds.
+
+        - 'lag' : like time, but past the halfway point counts as negative values.
+
+        - 'lag_s' : same as lag, but in seconds.
+
+        - 'lag_ms' : same as lag, but in milliseconds.
+
+        - `None`, 'none', or 'off': ticks and tick markers are hidden.
+
+
+    ax : matplotlib.axes.Axes or None
+        Axes to plot on instead of the default `plt.gca()`.
+
+    offset : float
+        Horizontal offset (in seconds) to start the waveform plot
+
+    marker : string
+        Marker symbol to use for sample values. (default: no markers)
+
+    where : string, {'pre', 'mid', 'post'}
+        This setting determines how both waveform and envelope plots interpolate
+        between observations.
+
+        See `matplotlib.pyplot.step` for details.
+
+        Default: 'post'
```

Comment:
what is the reason why we default to `post` if plt.step defaults to `pre`?

Reply:
Great question.  I actually have no idea why step defaults to `pre` when it's obviously wrong :grin: 

Here's how it looks with post:
![image](https://user-images.githubusercontent.com/1190540/118401016-0432f300-b632-11eb-8ec4-1745fa80dc58.png)

and here's how it looks with pre:
![image](https://user-images.githubusercontent.com/1190540/118401038-1ad94a00-b632-11eb-812e-be74900a71e3.png)

Reply:
... let's keep `post`